### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by epnev**
VolPy and ring-CNN

### CODE REVIEW

File: caiman/motion_correction.py [hunk 1]
```diff
@@ -2706,10 +2706,15 @@ def motion_correct_batch_rigid(fname, max_shifts, dview=None, splits=56, num_spl
             save_movie = save_movie_rigid
             logging.debug('saving!')
 
+
+        if isinstance(fname, tuple):
+            base_name=os.path.split(fname[0])[-1][:-4] + '_rig_'
```
Comment (pgunn) [inline caiman/motion_correction.py]: Could use a comment to make it clear what this (not very legible) transformation is doing

File: caiman/motion_correction.py [hunk 2]
```diff
@@ -2826,13 +2831,23 @@ def motion_correct_batch_pwrigid(fname, max_shifts, strides, overlaps, add_to_mo
         if iter_ == num_iter - 1:
             save_movie = save_movie
             if save_movie:
-                logging.debug('saving mmap of ' + fname)
+
+                if isinstance(fname, tuple):
+                    logging.debug('saving mmap of ' + fname[0] + 'to' +fname[-1])
```
Comment (pgunn) [inline caiman/motion_correction.py]: f-string

File: caiman/source_extraction/cnmf/initialization.py [hunk 3]
```diff
@@ -613,7 +613,15 @@ def compressedNMF(Y_ds, nr, r_ov=10, max_iter_snmf=500,
     T, dims = m.shape[0], m.shape[1:]
     d = np.prod(dims)
     yr = np.reshape(m, [T, d], order='F')
-    A, C, USV = nnsvd_init(yr, nr, r_ov=r_ov)    
+#    L = randomized_range_finder(yr, nr + r_ov, 3)
```
Comment (pgunn) [inline caiman/source_extraction/cnmf/initialization.py]: Why are we adding commented-out code?

File: caiman/source_extraction/cnmf/online_cnmf.py [hunk 4]
```diff
@@ -1085,12 +1089,47 @@ def fit_online(self, **kwargs):
         Returns:
             self (results of caiman online)
         """
-
+        self.t_init = -time()
```
Comment (pgunn) [inline caiman/source_extraction/cnmf/online_cnmf.py]: why is this negative?

File: caiman/source_extraction/cnmf/online_cnmf.py [hunk 5]
```diff
@@ -1120,6 +1159,9 @@ def fit_online(self, **kwargs):
                                   True)
         # Iterate through the epochs
         for iter in range(epochs):
+            if iter == epochs - 1 and self.params.get('online', 'stop_detection'):
+                self.params.set('online', {'update_num_comps': False})
+            logging.info('Searching for new components set to: {}'.format(self.params.get('online', 'update_num_comps')))
```
Comment (pgunn) [inline caiman/source_extraction/cnmf/online_cnmf.py]: f-string

File: caiman/source_extraction/cnmf/initialization.py (hunk 3 above)
Comment (epnev) [reply caiman/source_extraction/cnmf/initialization.py]: wip

**Pull request merged**